src/create-boilerplate.ts

```diff
@@ -0,0 +1,71 @@
+import { PromptObject } from 'prompts';
+import { readApiData, writeApiData, promptQuestions } from './utils/filesystem';
+import { runAndHandleErrors } from './utils/cli';
+import { emptyObject } from './utils/normalization';
+import { Api, ApiMetadata, Beacons, Oises, Templates } from './types';
+
+const questions: PromptObject[] = [
+  {
+    type: 'text',
+    name: 'apiName',
+    message: 'What is the name of the API Integration?',
+  },
+  {
+    type: 'text',
+    name: 'apiDescription',
+    message: 'Description of the API Integration',
+  },
+  {
+    type: 'text',
+    name: 'apiContact',
+    message: 'Contact email for the API Integration',
+  },
+];
+
+const main = async () => {
+  const response = await promptQuestions(questions);
+  const apiDataTemplate = readApiData();
+
+  // Create the boilderplate apiMetadata
+  const apiMetadata = emptyObject(apiDataTemplate.apiMetadata, ['active'], []) as ApiMetadata;
+  const apiMetadataBoilerplate = {
+    ...apiMetadata,
+    name: response.apiName,
+    contact: response.apiContact,
+    description: response.apiDescription,
+  } as ApiMetadata;
+
+  // Create the boilderplate ois
+  const oises = emptyObject(
+    apiDataTemplate.ois,
+    ['oisFormat', 'version'],
+    ['paths', 'fixedOperationParameters', 'reservedParameters', 'parameters']
+  ) as Oises;
+  const oisBoilerPlate = { [response.apiName + '-' + '1.0.0.json']: oises[Object.keys(oises)[0]] } as Oises;
+
+  // Create the boilerplate template
+  const templates = emptyObject(apiDataTemplate.templates, [], ['decodedParameters']) as Templates;
+  const templateBoilerPlate = {
+    [response.apiName + '-' + 'tempate01.json']: templates[Object.keys(templates)[0]],
```

Comment:
Spelling :)

Reply:
Also, these object keys become file names and not all OSes/filesystems support case sensitivity (like Mac OS, which can be either case sensitive or case insensitive), so keys that become filenames should be filename santised:
```
sanitiseFilename(response.apiName + '-' + 'tempate01.json')
```